go/src/staleFlagCleaner.go

```diff
@@ -0,0 +1,624 @@
+					} else {
+						//default case
+						//here assuming defualt case is at last
+						if notfoundCase {
+							for _, bodyEle := range exprstmt.(*dst.CaseClause).Body {
+								n.InsertBefore(bodyEle)
+							}
+						}
+					}
+				}
+				n.Delete()
+			}
+		} else {
+			//now switch is like elseif statements so do accordingly
+			needDeletion := false
+			isBotAbove := false
+			for ind := 0; ind < len(d.Body.List); ind++ {
+				exprstmt := d.Body.List[ind]
+				if exprstmt.(*dst.CaseClause).List != nil {
+					valCaseClause := sfc.evaluateExprNode(exprstmt.(*dst.CaseClause).List[0])
+					if valCaseClause == isTrue {
+						//this has to go out as switch is going to delete
+						//On the condition that there was no expression above whose value is isBot.
+						if !isBotAbove {
+							for _, bodyEle := range exprstmt.(*dst.CaseClause).Body {
+								n.InsertBefore(bodyEle)
+							}
+							needDeletion = true
+							break
+						} else {
+							d.Body.List = d.Body.List[:ind+1]
+							d.Body.List[ind].(*dst.CaseClause).List = nil
+							break
+						}
+					} else if valCaseClause == isFalse {
+						if len(d.Body.List) > 1 {
+							copy(d.Body.List[ind:], d.Body.List[ind+1:])   // Shift d.Body.List[ind+1:] left one index.
+							d.Body.List[len(d.Body.List)-1] = nil          // Erase last element (write zero value).
+							d.Body.List = d.Body.List[:len(d.Body.List)-1] // Truncate slice.
+
+							ind = ind - 1
+						} else {
+							n.Delete()
+							break
+						}
+					} else {
+						// so now value is bot
+						isBotAbove = true
+					}
+				}
+			}
+			if needDeletion {
+				n.Delete()
+			}
+		}
+	//handling for in general experssions
+	case *dst.BinaryExpr:
+		valX := sfc.evaluateExprNode(d.X)
+		valY := sfc.evaluateExprNode(d.Y)
+		if valX == isBot || valY == isBot {
+			if valX == isBot && valY == isBot {
+				//do nothing
+			} else {
+				//now one of them is not bot
+				//check for valX
+				if valX == isTrue && d.Op != token.LOR {
+					n.Replace(d.Y)
+				}
+				if valX == isFalse && d.Op != token.LAND {
+					n.Replace(d.Y)
+				}
+				if valY == isTrue && d.Op != token.LOR {
+					n.Replace(d.X)
+				}
+				if valY == isFalse && d.Op != token.LAND {
+					n.Replace(d.X)
+				}
+				if d.Op == token.EQL {
+					newNode := n.Node()
+					var trueIdent dst.Ident
+					// either valX is not bot or valY. both cannot be bot at the same time as we checked out above
+					if valX != isBot {
+						if valX == isTrue {
+							trueIdent.Name = strTrue
+						} else {
+							trueIdent.Name = strFalse
+						}
+						newNode.(*dst.BinaryExpr).X = dst.Expr(&trueIdent)
+						n.Replace(newNode)
+					} else {
+						if valY == isTrue {
+							trueIdent.Name = strTrue
+						} else {
+							trueIdent.Name = strFalse
+						}
+						newNode.(*dst.BinaryExpr).Y = dst.Expr(&trueIdent)
+						n.Replace(newNode)
+					}
+				}
+			}
+		}
+	default:
+
+	}
+	return true
+}
+
+// This is the Upper flow of the programme
```

Comment:
According to our style guide, you order your file (roughly) in the calling order. It makes sense in general, as it improves readability - you start with the "main" part and look for the helper functions as you keep parsing the code, rather than the other way around, that is starting with looking at helper functions.